### PR TITLE
Refactor implementation of the various RUN flavors

### PR DESCRIPTION
**Pull request by vladaionescu**
* Unify all implementations of run into a single big, general function
* Ensures `LOCALLY` commands run sequentially (Fixes https://github.com/earthly/earthly/issues/938, part of https://github.com/earthly/earthly/issues/580)
* Removes support for `RUN --push` under `LOCALLY`

### CODE REVIEW

**Review by dchw — APPROVED** [review]
Hard to follow the diffs! BUt I like pushing it all into `internalRun` more or less to keep behavior consistent.

I feel like we should get some unit tests around this function going forward.

**Review by alexcb — APPROVED** [review]

File: earthfile2llb/converter.go [hunk 1]
```diff
@@ -1517,21 +1439,43 @@ func (c *Converter) internalRun(ctx context.Context, args, secretKeyValues []str
...
 }
 
+func (c *Converter) forceExecution(ctx context.Context, state pllb.State) error {
+	ref, err := llbutil.StateToRef(ctx, c.opt.GwClient, state, c.opt.Platform, c.opt.CacheImports.AsMap())
+	if err != nil {
+		return errors.Wrap(err, "run locally state to ref")
+	}
+	// We're not really interested in reading the dir - we just
+	// want to un-lazy the ref so that the local commands have executed.
+	_, err = ref.ReadDir(ctx, gwclient.ReadDirRequest{Path: "/"})
```
Comment (alexcb) [inline earthfile2llb/converter.go:1472]: nice hack :)
